File: web/web-frontend/src/app/model/text/parts/param-meta/boolean-type.meta.ts
import {TypeMeta} from "./type-meta.model";
import {Serializable} from "../../../infrastructure/serializable.model";
import {JsonUtil} from "../../../../utils/json.util";

export class BooleanTypeMeta implements TypeMeta, Serializable<BooleanTypeMeta> {
    javaType: string;

    constructor(javaType = "java.lang.Boolean") {
        this.javaType = javaType;
    }

    deserialize(input: Object): BooleanTypeMeta {
        this.javaType = input["javaType"];
        return this;
    }

    serialize(): string {
        return ''+
               '{' +
               '"@type": "BOOLEAN"'+
               ',"javaType":'+ JsonUtil.stringify(this.javaType) +
               '}'
    }
}

File: model/src/main/kotlin/com/testerum/model/text/parts/param_meta/BooleanTypeMeta.kt
package com.testerum.model.text.parts.param_meta

import com.fasterxml.jackson.annotation.JsonProperty

class BooleanTypeMeta(@JsonProperty("javaType") override val javaType: String = "java.lang.Boolean") : TypeMeta {
    override fun fileType(): String {
        return "BOOLEAN";
    }
}
